src/components/shared/chart/Canvas.js

```diff
@@ -128,18 +129,31 @@ export default class ChartCanvas<HoveredItem> extends React.Component<
     }
   }
 
-  _onMouseDown = (
-    event: { nativeEvent: MouseEvent } & SyntheticMouseEvent<>
-  ) => {
-    // Remember where the mouse was positioned. Move too far and it
-    // won't be registered as a selecting click on mouse up.
-    this._mouseDownOffsetX = event.nativeEvent.offsetX;
-    this._mouseDownOffsetY = event.nativeEvent.offsetY;
-    this._mouseMovedWhileClicked = false;
+  _onMouseDown = (e: { nativeEvent: MouseEvent } & SyntheticMouseEvent<>) => {
+    if (e.button === 0) {
+      // Remember where the mouse was positioned. Move too far and it
+      // won't be registered as a selecting click on mouse up.
+      this._mouseDownOffsetX = e.nativeEvent.offsetX;
+      this._mouseDownOffsetY = e.nativeEvent.offsetY;
+      this._mouseMovedWhileClicked = false;
+    }
+
+    if (e.button === 2 && this.props.onRightClick) {
+      // The right button is a contextual action.
+      // It is important that we call the right click callback at mousedown so
+      // that the state is updated and the context menus are rendered before the
+      // mouseup/contextmenu events.
+      this.props.onRightClick(this.state.hoveredItem);
+    }
```

Comment:
I wonder if we should call `onSelectItem` if there's no `onRightClick`, like I do in the TreeView.